scipy/signal/filter_design.py

```diff
@@ -3653,6 +3653,222 @@ def besselap(N, norm='phase'):
+    .. versionadded: 0.19.0
+
+    References
+    ----------
+    .. [1] Sophocles J. Orfanidis, "Introduction To Signal Processing",
+           Prentice-Hall, 1996
+
+    Examples
+    --------
+    Design and plot filter to remove the 60Hz component from a
+    signal sampled at 200Hz, using a quality factor Q = 30
+
+    >>> from scipy import signal
+    >>> import numpy as np
+    >>> import matplotlib.pyplot as plt
+
+    >>> fs = 200.0  # Sample frequency (Hz)
+    >>> f0 = 60.0  # Frequency to be removed from signal (Hz)
+    >>> Q = 30.0  # Quality factor
+    >>> w0 = f0/(fs/2)  # Normalized Frequency
+    >>> # Design notch filter
+    >>> b, a = signal.iirnotch(w0, Q)
+
+    >>> # Frequency response
+    >>> w, h = signal.freqz(b, a)
+    >>> # Generate frequency axis
+    >>> freq = w*fs/(2*np.pi)
+    >>> # Plot
+    >>> fig, ax = plt.subplots(2, 1, figsize=(8, 6))
+    >>> ax[0].plot(freq, 20*np.log10(abs(h)), color='blue')
+    >>> ax[0].set_title("Frequency Response")
+    >>> ax[0].set_ylabel("Amplitude (dB)", color='blue')
+    >>> ax[0].set_xlim([0, 100])
+    >>> ax[0].set_ylim([-25, 10])
+    >>> ax[0].grid()
+    >>> ax[1].plot(freq, np.unwrap(np.angle(h))*180/np.pi, color='green')
+    >>> ax[1].set_ylabel("Angle (degrees)", color='green')
+    >>> ax[1].set_xlabel("Frequency (Hz)")
+    >>> ax[1].set_xlim([0, 100])
+    >>> ax[1].set_yticks([-90, -60, -30, 0, 30, 60, 90])
+    >>> ax[1].set_ylim([-90, 90])
+    >>> ax[1].grid()
+    >>> plt.show()
+    """
+
+    # Guarantee that the inputs are floats
+    w0 = float(w0)
+    Q = float(Q)
+
+    # Checks if w0 is within the range
+    if w0 > 1.0 or w0 < 0.0:
+        raise ValueError("w0 should be such that 0 < w0 < 1")
+
+    # Get bandwidth
+    bw = w0/Q
+
+    # Normalize inputs
+    bw = bw*np.pi
+    w0 = w0*np.pi
+
+    # Compute -3dB atenuation
+    gb = 1/np.sqrt(2)
+
+    # Compute beta: formula 11.3.4 (p.575) from reference [1]
+    beta = (np.sqrt(1.0-gb**2.0)/gb)*np.tan(bw/2.0)
+
+    # Compute gain: formula 11.3.6 (p.575) from reference [1]
+    gain = 1.0/(1.0+beta)
+
+    # Compute numerator b and denominator a
+    # formula 11.3.7 (p.575) from reference [1]
+    b = gain*np.array([1.0, -2.0*np.cos(w0), 1.0])
+    a = np.array([1.0, -2.0*gain*np.cos(w0), (2.0*gain-1.0)])
+
+    return b, a
+
+
+def iirpeak(w0, Q):
+    """
+    Design second-order IIR peak (resonant) digital filter.
+
+    A peak filter is a band-pass filter with a narrow bandwidth
+    (high quality factor). It rejects components outside a narrow
+    frequency band.
+
+    Parameters
+    ----------
+    w0 : float
+        Normalized frequency to be retained in a signal. It is a
+        scalar that must satisfy  ``0 < w0 < 1``, with ``w0 = 1`` corresponding
+        to half of the sampling frequency.
+    Q : float
+        Quality factor. Dimensionless parameter that characterizes
+        peak filter bandwidth ``bw`` relative to its center
+        frequency, ``Q = w0/bw``.
```

Comment:
It may be helpful here to mention that ``bw`` is the 3dB bandwidth, now that `rp` is effectively fixed there.